clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputUtils.java

```diff
@@ -55,6 +72,35 @@ public static VeniceProperties getConsumerProperties(JobConf config) {
     return new VeniceProperties(consumerFactoryProperties);
   }
 
+  public static KafkaValueSerializer getKafkaValueSerializer(JobConf config) {
+    KafkaValueSerializer kafkaValueSerializer = new OptimizedKafkaValueSerializer();
+    boolean isSchemaReaderEnabled = Boolean.parseBoolean(config.get(SCHEMA_READER_ENABLED, "false"));
```

Comment:
I saw this is mainly for KIF repush, which means the d2 host/service needs to align with the repush source colo, right?
Having them as hard-coded config could introduce some maintenance burden, and if we need to configure a different zk host when switching repush source, that would introduce some operation toils.

If the above assumption is true, I would like to automate this and here is some idea:
1. Talk to parent controller to retrieve the zk host for a particular region.
2. We can pass the region param in the call to Parent Controller.
3. Parent Controller will talk to the corresponding Child Controller to retrieve these setting.

Or an even simpler way is to return the ZK host/service in the `RepushInfoResponse`, which is region specific:
https://github.com/linkedin/venice/blob/main/internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java#L242

The `RepushInfoResponse` already contains the Kafka url for the source region and we can add source ZK into it, so that there is no ZK config for repush and it is fully automated.

What do you think?

Reply:
This is a great idea to reduce maintenance overhead. Thanks. I made controller changes to return system schema cluster d2 service name and zk host of the source olo in `RepushInfoResponse`.